test/test_bmp.cpp

```diff
@@ -0,0 +1,319 @@
+#include <catch2/catch_all.hpp>
+#include "../src/bmp.hpp"
+#include "../src/pixel.hpp"
+
+#include <bit>
+#include <iostream>
+
+using Catch::Matchers::ContainsSubstring;
+
+TEST_CASE("Read BMP files" , "[bmp][read]") {
+    std::string bmp_path;
+    SECTION("Reads solid_white.bmp successfully") {
+        bmp_path = "test_data/solid_white.bmp";
+        BMP solid_white;
+
+        REQUIRE_NOTHROW(solid_white.read(bmp_path.c_str()));
```

Comment:
* Instead of using `BMP::read`, why not use the corresponding `BMP` constructor?  (Because that can't easily be put into a `REQUIRE_NOTHROW`?  Then see the Catch2 documentation and use a lambda.)
* I'd de-clutter this and directly pass a string literal into the `read` call, getting rid of `bmp_path` and `.c_str()`